src/main/views/View.java

```diff
@@ -0,0 +1,50 @@
+package main.views;
+
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.Scene;
+import javafx.scene.layout.Pane;
+import javafx.stage.Screen;
+
+import java.io.IOException;
+
+public abstract class View {
+    /* Literally a number that I choose without any critical thinking */
+    private static final double BOUND_FACTOR = 1.8;
+    /* Primary Screen (useful if the user has multiple monitors bounds */
+    private final Rectangle2D primaryScreenBounds = Screen.getPrimary().getVisualBounds();
+    private Scene scene;
+    /* *
+     * Pane is extended from all Parent object in fxml files that we have,
+     * it makes sense to use it as a common field type
+     * */
+    private Pane parent;
+
+    Scene getScene() {
+        return this.scene;
+    }
+
+    Pane getParent() {
+        return this.parent;
+    }
+
+    double getWidth() {
+        return this.primaryScreenBounds.getWidth() / BOUND_FACTOR;
+    }
+
+    double getHeight() {
+        return primaryScreenBounds.getHeight() / BOUND_FACTOR;
+    }
+
+    void setScene() {
+        this.scene = new Scene(this.parent, getWidth(), getHeight());
+    }
+
+    void setParent(String filename) {
+        try {
+            this.parent = FXMLLoader.load(getClass().getResource(filename));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.

Reply:
That's true, thanks for the suggestion, good bot <3

Reply:
Ok, I will recheck. Thanks a lot!

Reply:
Bot, check it in my next pull request